zipline/pipeline/filters/filter.py

```diff
@@ -427,3 +428,29 @@ class Latest(LatestMixin, CustomFilter):
     Filter producing the most recently-known value of `inputs[0]` on each day.
     """
     pass
+
+
+class SingleAsset(Filter):
+    """
+    A Filter that computes to True only for the given asset.
+    """
+    inputs = []
+    window_length = 1
+
+    def __new__(cls, asset):
+        return super(SingleAsset, cls).__new__(cls, asset=asset)
+
+    def _init(self, asset, *args, **kwargs):
+        self._asset = asset
+        return super(SingleAsset, self)._init(*args, **kwargs)
+
+    @classmethod
+    def static_identity(cls, asset, *args, **kwargs):
+        return (
+            super(SingleAsset, cls).static_identity(*args, **kwargs), asset,
+        )
+
+    def _compute(self, arrays, dates, assets, mask):
```

Comment:
Notes:

0. This shouldn't mutate the mask in place.  The input arrays here are shared with other computations, so mutating them will break all kinds of things and make pipeline results non-deterministic on the order in which they're computed.

1. You don't actually need the index to build the mask.  You can just do `zipline.utils.numpy_utils.repeat_first_axis(assets == self._asset.sid)`.

e.g.:
```
In [14]: sids = arange(5)

In [15]: sids
Out[15]: array([0, 1, 2, 3, 4])

In [16]: sid = 3

In [17]: sids == sid
Out[17]: array([False, False, False,  True, False], dtype=bool)

In [18]: repeat_first_axis(sids == sid, 10)
Out[18]:
array([[False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False],
       [False, False, False,  True, False]], dtype=bool)
```

2. We still need to assert that the asset is in `assets`.

The right implementation of this is probably something like:

```
def _compute(self, arrays, dates, assets, mask):
    is_my_asset = (assets == self._asset.sid)
    if is_my_asset.sum() != 1:
        raise YourAssetDidntExistYet(...)  # Probably need a better name for this.
    out = repeat_first_axis(is_my_asset, len(mask))
    # Maybe assert that mask is True everywhere that out is?
    # i.e. assert (out & mask).sum() == len(mask)
    return out
```